src/instruments/src/PFD/AltitudeIndicator.tsx

```diff
@@ -106,6 +107,99 @@ class RadioAltIndicator extends DisplayComponent<{ bus: EventBus, filteredRadioA
     }
 }
 
+class MinimumDescentAltitudeIndicator extends DisplayComponent<{ bus: EventBus }> {
+    private visibility = Subject.create('hidden');
+
+    private path = Subject.create('');
+
+    private altitude = 0;
+
+    private radioAltitudeValid = false;
+
+    private qnhLandingAltValid = false;
+
+    private qfeLandingAltValid = false;
+
+    private inLandingPhases = false;
+
+    private altMode: 'STD' | 'QNH' | 'QFE' = 'STD';
+
+    private mda = new Arinc429Word(0);
+
+    private landingElevation = new Arinc429Word(0);
+
+    private updateIndication(): void {
+        this.qnhLandingAltValid = !this.landingElevation.isFailureWarning()
+            && !this.landingElevation.isNoComputedData()
+            && this.inLandingPhases
+            && this.altMode === 'QNH';
+
+        this.qfeLandingAltValid = this.inLandingPhases
+            && this.altMode === 'QFE';
+
+        const altDelta = this.mda.value - this.altitude;
+
+        const showMda = (this.radioAltitudeValid || this.qnhLandingAltValid || this.qfeLandingAltValid)
+            && Math.abs(altDelta) <= 570
+            && !this.mda.isFailureWarning()
+            && !this.mda.isNoComputedData();
+
+        if (!showMda) {
+            this.visibility.set('hidden');
+            return;
+        }
+
+        const offset = altDelta * DistanceSpacing / ValueSpacing;
+        this.path.set(`m 127.9276,${80.249604 - offset} h 5.80948 v 1.124908 h -5.80948 z`);
+        this.visibility.set('visible');
+    }
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        const sub = this.props.bus.getSubscriber<PFDSimvars & Arinc429Values & SimplaneValues>();
+
+        sub.on('chosenRa').whenArinc429SsmChanged().handle((ra) => {
+            this.radioAltitudeValid = !ra.isFailureWarning() && !ra.isNoComputedData();
+            this.updateIndication();
+        });
+
+        sub.on('landingElevation').withArinc429Precision(0).handle((landingElevation) => {
+            this.landingElevation = landingElevation;
+            this.updateIndication();
+        });
+
+        sub.on('baroMode').whenChanged().handle((m) => {
+            this.altMode = m;
+            this.updateIndication();
+        });
+
+        sub.on('altitudeAr').withArinc429Precision(0).handle((a) => {
+            // TODO filtered alt
+            this.altitude = a.value;
+            this.updateIndication();
+        });
+
+        sub.on('mda').handle((mda) => {
```

Comment:
I don't think that values changes very often. Consider using `.whenChanged()`

Reply:
Good catch! 😀